src/blog/2024/01/unified-namespace-what-broker.md

```diff
@@ -0,0 +1,50 @@
+subtitle: The broker is the data backbone for the unified namespace, which one is right for you?
+description: The broker is the data backbone for the unified namespace, which one is right for you?
+date: 2024-01-16
+authors: ["zeger-jan-van-de-weg"]
+image: # TODO
+tags:
+    - posts
+    - flowfuse
+---
+
+When starting to roll out a new data distribution architecture for the unified namespace (UNS), one of the first question you'll ask is "What broker should I select for my UNS? The broker must implement a publish-subscribe (pub sub) pattern, though that leaves plenty of options.
+
+<!--more-->
+
+## Technology selection
+
+### Two protocols frontrunners
+
+Currently there’s two protocols that are front runners for becoming the de facto data transfer choice in (industrial) IoT: MQTT or Kafka. They’ve been designed for different use-cases, and have different properties. Currently MQTT is more often deployed as a broker in the unified namespace, and is generally the best choice when starting to implement a unified namespace.
+
+First and foremost, MQTT has been designed to enable IoT use-cases. Main design objectives were to be lightweight to enable low-bandwidth communication, enable low-power devices, and handle unreliable networks. MQTT enables a large number of data producers and consumers to collaborate.
+
+Kafka is designed as an event streaming platform. And its initial adoption was mostly for data brokers between microservices all part of the same web-backend for large sites like LinkedIn. When communicating data between servers or just a few data centers around the world, there’s less of a concern around the reliability of the connection, or to enable constrained devices to participate in the shift towards a unified namespace.
+
+Generally MQTT is more often seen deployed in practice as data distribution architecture.
+
+
+### The Cloud route
+
+Cloud message queue brokers like AWS Kinesis and GCP Pub/Sub offer a high level of convenience. Scaling the infrastructure for real-time data processing and communication is their concern, the customer is mostly concerned about paying the bill. These brokers are fully managed, meaning they are maintained and updated by the cloud provider, reducing the burden on developers.
+
+However, this convenience comes with the tradeoff of vendor lock-in. When selecting these brokers, the cloud vendor has usually adapted their technology to support many protocols, and these offerings are usually jack-off-all-trades solutions – master of none. It creates a situation where the unified namespace implementation will change in subtle ways to accommodate the vendor, instead of the other way around. An organization might become so reliant on a particular vendor's products or services that they are unable to easily switch to another vendor or protocol which serves their business objectives better. The cost of changing is exacerbated by having to train personnel on new and open-source protocols.
```

Comment:
```suggestion
However, this convenience comes with the tradeoff of vendor lock-in. When selecting these brokers, the cloud vendor has usually adapted their technology to support many protocols, and these offerings are usually jack-of-all-trades solutions – master of none. It creates a situation where the unified namespace implementation will change in subtle ways to accommodate the vendor instead of the other way around. An organization might become so reliant on a particular vendor's products or services that they are unable to easily switch to another vendor or protocol that serves their business objectives better. The cost of changing is exacerbated by having to train personnel on new and open-source protocols.
```